dashboard/app/models/contact_rollups_raw.rb

```diff
@@ -33,6 +88,36 @@ def self.extract_parent_emails
     ContactRollupsV2.execute_query_in_transaction(query)
   end
 
+  def self.extract_scripts_taught
+    source_sql = <<~SQL
+      SELECT email, scripts.name, MAX(se.updated_at) AS updated_at
+      FROM users u
+      JOIN sections se ON se.user_id = u.id
+      JOIN scripts ON scripts.id = se.script_id
+      WHERE scripts.name in (#{CSF_SCRIPT_LIST})
+      AND email > ''
+      GROUP BY email, scripts.name
+    SQL
+
+    query = get_extraction_query(source_sql, 'email', ['name'], true, 'dashboard.sections.script_id')
+    ContactRollupsV2.execute_query_in_transaction(query)
+  end
+
+  def self.extract_courses_taught
+    source_sql = <<~SQL
+      SELECT email, courses.name, MAX(se.updated_at) AS updated_at
+      FROM users u
+      JOIN sections se ON se.user_id = u.id
+      JOIN courses ON courses.id = se.course_id
+      WHERE courses.name in (#{COURSES_TAUGHT_LIST})
```

Comment:
Same as above, can we get all courses first (in the extraction step), and categorize them later (in the processing step).